packages/@uppy/companion/src/server/controllers/s3.js

```diff
@@ -35,13 +35,16 @@ module.exports = function s3 (config) {
     // @ts-ignore The `companion` property is added by middleware before reaching here.
     const client = req.companion.s3Client
 
-    if (!client || typeof config.bucket !== 'string') {
+    if (!client || (typeof config.bucket !== 'string' && typeof config.bucket !== 'function')) {
       res.status(400).json({ error: 'This Companion server does not support uploading to S3' })
       return
     }
 
     const metadata = req.query.metadata || {}
     const key = config.getKey(req, req.query.filename, metadata)
+
+    const bucket = typeof config.bucket === 'function' ? config.bucket(req) : config.bucket
```

Comment:
Can you make a util `getBucket`?
```suggestion
    const bucket = getBucket(config.bucket, req)
```

Reply:
for sure! when u guys mentioned reusability i was thinking about these in first place.
also was thinking if would be better to keep a bucket string only and having a similar getKey but for bucket, getBucket in this case for the config options

Reply:
That could be useful indeed, good call.